src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementEventNormalizationService.cs

```diff
@@ -115,19 +115,7 @@ private async Task StartConsumer(ISourceBlock<EventData> producer, IEnumerableAs
 
                         var token = _converter.Convert(evt);
 
-                        try
-                        {
-                            foreach (var measurement in _contentTemplate.GetMeasurements(token))
-                            {
-                                measurement.IngestionTimeUtc = evt.SystemProperties.EnqueuedTimeUtc;
-                                createdMeasurements.Add((partitionId, measurement));
```

Comment:
Is there a reason we can't just yield return here?

Reply:
Unfortunately a `yield` statement cannot existing inside of a `try` block if there is a `catch` block.

https://docs.microsoft.com/en-us/archive/msdn-magazine/2017/june/essential-net-custom-iterators-with-yield#yield-statement-requirements